crates/ruma-events/src/image_pack.rs

```diff
@@ -0,0 +1,160 @@
+//! Types for image packs in Matrix ([MSC2545]).
+//!
+//! [MSC2545]: https://github.com/matrix-org/matrix-spec-proposals/pull/2545
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use ruma_common::{serde::StringEnum, OwnedMxcUri, OwnedRoomId};
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+
+use crate::{room::ImageInfo, PrivOwnedStr};
+
+/// The content of an `im.ponies.room_emotes` event,
+/// the unstable version of `m.image_pack` in room state events.
+///
+/// State key is the identifier for the image pack in [ImagePackRoomsEventContent].
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[ruma_event(type = "im.ponies.room_emotes", kind = State, state_key_type = String)]
+pub struct RoomImagePackEventContent {
+    /// A list of images available in this image pack.
+    ///
+    /// Keys in the map are shortcodes for the images.
+    pub images: BTreeMap<String, PackImage>,
+    /// Image pack info.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pack: Option<PackInfo>,
+}
+
+impl RoomImagePackEventContent {
+    /// Creates a new empty `RoomImagePackEventContent`.
+    pub fn new() -> Self {
```

Comment:
The constructor should actually take `images` as a parameter because it is a required field. Same for `AccountImagePackEventContent`.